provider/env.go

```diff
@@ -0,0 +1,42 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func envResource() *schema.Resource {
+	return &schema.Resource{
+		Description: "Use this resource to set an environment variable in a workspace.",
+		CreateContext: func(_ context.Context, rd *schema.ResourceData, _ interface{}) diag.Diagnostics {
+			rd.SetId(uuid.NewString())
+
+			return nil
+		},
+		ReadContext:   schema.NoopContext,
+		DeleteContext: schema.NoopContext,
+		Schema: map[string]*schema.Schema{
+			"agent_id": {
+				Type:        schema.TypeString,
+				Description: `The "id" property of a "coder_agent" resource to associate with.`,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The name of the environment variable.",
+				ForceNew:    true,
+				Required:    true,
+			},
+			"value": {
```

Comment:
Thinking loud: `default_value` if not set?

Reply:
Something similar to: using generic dot files if the user didn't provide a custom URL. But I agree to postpone it for later until we meet scenario.

Reply:
This is fine for now, thanks for challenging